### PR TITLE
Improve validation of ParameterConfigs from config file

### PR DESCRIPTION
**Pull request by eivindjahren**
## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [x] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/config/surface_config.py [hunk 1]
```diff
@@ -18,16 +22,72 @@
...
+    xinc: float
+    yinc: float
+    rotation: float
     yflip: int
     forward_init_file: str
     output_file: Path
     base_surface_path: str
 
+    @classmethod
+    def from_config_list(cls, surface: List[str]) -> Self:
```
Comment (yngve-sk) [inline src/ert/config/surface_config.py:36]: Does `surface` always have context? In that case it is a `ContextList[str]`

File: src/ert/config/gen_kw_config.py [hunk 2]
```diff
@@ -41,6 +62,108 @@ def __post_init__(self) -> None:
         for e in self.transfer_function_definitions:
             self.transfer_functions.append(self._parse_transfer_function(e))
 
+        self._validate()
+
+    @classmethod
+    def from_config_list(cls, gen_kw: List[str]) -> Self:
+        gen_kw_key = gen_kw[0]
```
Comment (yngve-sk) [inline src/ert/config/gen_kw_config.py:69]: Why are some errors here and some in _validate?

File: src/ert/config/ensemble_config.py [hunk 3]
```diff
@@ -172,50 +110,17 @@ def __init__(  # noqa: 501 pylint: disable=too-many-arguments, too-many-branches
...
-            if len(all_arglists) > 1:
-                raise ConfigValidationError.with_context(
-                    f"Found two GEN_KW {kw_id} declarations", kw_id
-                )
-
-            return next(
-                (arg for arg in all_arglists[0] if matching.lower() in arg.lower()),
-                None,
+    @staticmethod
+    def _validate(gen_kw_list: List[GenKwConfig], gen_data_list: List[GenDataConfig]):
```
Comment (yngve-sk) [inline src/ert/config/ensemble_config.py:109]: This checks only that no duplicate names exist across the gen kws and gen data, then `_validate` suggests it does a bit more than it really does

File: src/ert/config/gen_kw_config.py (hunk 2 above)
Comment (eivindjahren) [reply src/ert/config/gen_kw_config.py:69]: Errors in `from_config_list` should be errors related to making the string in config into the parameters of the `GenKw` constructor. The errors in `GenKw` constructor (`_validate`) checks whether you have a valid object, ie. it relates to the transaction:

```python
try:
    genkw = GenKwConfig(...)
catch Exception err:
    # here, genkw is not initialized
# here you have a valid genkw object.

File: src/ert/config/surface_config.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/config/surface_config.py:36]: The context for the surface is always the `List[str]`. Which is actually `ContextList[FileContextToken]`

File: src/ert/config/ensemble_config.py (hunk 3 above)
Comment (eivindjahren) [reply src/ert/config/ensemble_config.py:109]: That is the only validation for `EnsembleConfig`, but we could just rename it :)

Comment (eivindjahren) [reply src/ert/config/ensemble_config.py]: Renamed it :)

**Review by yngve-sk — APPROVED** [review]
🟩

**Pull request merged**